### PR TITLE
feat: Security Rules Management API

### PR DESCRIPTION
**Pull request by hiranya911**
RELEASE NOTE: Added a new `admin.securityRules()` API for managing Firebase security rules applied on services like Cloud Firestore and Cloud Storage.
RELEASE NOTE: Added `getFirestoreRuleset()` and `getStorageRuleset()` methods for retrieving rulesets that are currently in effect.
RELEASE NOTE: Added `releaseFirestoreRuleset()` and `releaseStorageRuleset()` methods for creating new rulesets and applying them to Firestore and Storage.
RELEASE NOTE: Added `getRuleset()`, `createRuleset()` and `deleteRuleset()` methods for managing the lifecycle of a ruleset.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -363,6 +363,37 @@ declare namespace admin {
...
+  *
+  * `admin.securityRules()` can be called with no arguments to access the
+  * default app's {@link admin.securityRules.SecurityRules
+  * `SecurityRules`} service, or as `admin.securityRules(app)` to access
+  * the {@link admin.securityRules.SecurityRules `SecurityRules`}
+  * service associated with a specific app.
+  *
+  * @example
+  * ```javascript
+  * // Get the SecurityRules service for the default app
```
Comment (egilmorez) [inline src/index.d.ts]: Suggest backticks like you have below: "the `SecurityRules` service"

File: src/index.d.ts [hunk 2]
```diff
@@ -363,6 +363,37 @@ declare namespace admin {
...
+  * var defaultSecurityRules = admin.securityRules();
+  * ```
+  *
+  * @example
+  * ```javascript
+  * // Get the SecurityRules service for a given app
```
Comment (egilmorez) [inline src/index.d.ts]: Suggest backticks like you have below: "the `SecurityRules` service"

File: src/security-rules/security-rules.ts [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    }
+
+    this.name = stripProjectIdPrefix(ruleset.name);
+    this.createTime = new Date(ruleset.createTime).toUTCString();
+    this.source = ruleset.source.files || [];
+  }
+}
+
+/**
+ * SecurityRules service bound to the provided app.
```
Comment (egilmorez) [inline src/security-rules/security-rules.ts]: Suggest backticks like you have elsewhere: "the `SecurityRules` service"

File: src/security-rules/security-rules.ts [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+
+  private static readonly CLOUD_FIRESTORE = 'cloud.firestore';
+  private static readonly FIREBASE_STORAGE = 'firebase.storage';
+
+  public readonly INTERNAL = new SecurityRulesInternals();
+
+  private readonly client: SecurityRulesApiClient;
+
+  /**
+   * @param {object} app The app for this SecurityRules service.
```
Comment (egilmorez) [inline src/security-rules/security-rules.ts]: Suggest backticks like you have elsewhere: "the `SecurityRules` service"

File: src/index.d.ts [hunk 5]
```diff
@@ -363,6 +363,37 @@ declare namespace admin {
...
+  * var otherSecurityRules = admin.securityRules(otherApp);
+  * ```
+  *
+  * @param app Optional app to return the `SecurityRules` service
+  *     for. If not provided, the default `SecurityRules` service will
+  *     be returned.
```
Comment (egilmorez) [inline src/index.d.ts]: A good case for just "is returned."

File: src/index.d.ts [hunk 6]
```diff
@@ -5446,6 +5478,196 @@ declare namespace admin.projectManagement {
...
+    deleteRuleset(name: string): Promise<void>;
+
+    /**
+     * Retrieves a page of ruleset metadata.
+     *
+     * @param pageSize The page size, 100 if undefined. This is also the maximum allowed
+     *   limit.
+     * @param nextPageToken The next page token. If not specified, returns rulesets
+     *   starting without any offset.
+     * @return A promise that fulfills a page of rulesets.
```
Comment (egilmorez) [inline src/index.d.ts]: Should this be "fulfills with" ?

File: src/index.d.ts (hunk 1 above)
Comment (hiranya911) [reply src/index.d.ts]: Does it matter here? This is a comment inside an example. It would be rendered as is.

Comment (egilmorez) [reply src/index.d.ts]: Oh that's right :)

We can skip this one and the others inside comments for sure.

File: src/index.d.ts (hunk 5 above)
Comment (hiranya911) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 6 above)
Comment (hiranya911) [reply src/index.d.ts]: Done

**Comment by hiranya911** [comment]
Thanks @egilmorez. Made some of the changes suggested. As for adding backticks, I don't think it's needed there since they are rendered as comments in an example snippet. WDYT?

**Pull request merged**